cmd/gardener-admission-controller/app/gardener_admission_controller.go

```diff
@@ -174,6 +174,11 @@ func (o *options) run(ctx context.Context) error {
 	log.Info("setting up webhook server")
 	server := mgr.GetWebhookServer()
 
+	log.Info("setting up readycheck for webhook server")
+	if err := mgr.AddHealthzCheck("readyz", server.StartedChecker()); err != nil {
```

Comment:
This should be added to the readiness endpoint not the health endpoint.
```suggestion
	if err := mgr.AddReadyzCheck("webhook-server", server.StartedChecker()); err != nil {
```

Health endpoint is for checking if the container is still alive (or might need to be restarted). The readiness endpoint is for determining whether the pod should be added to the ready set of containers to serve traffic for the service (and for other things like continuing a rolling update). This is what we want for the case of webhooks.